service.go

```diff
@@ -38,15 +63,25 @@ func newLocationService(repo tmereader.Repository, baseURL string, taxonomyName
 }
 
 func (s *locationServiceImpl) getLocations() ([]locationLink, bool) {
-	if len(s.locationLinks) > 0 {
-		return s.locationLinks, true
+	val := s.locationLinks.Load()
+	if val == nil {
+		return nil, false
 	}
-	return s.locationLinks, false
+
+	tmp := val.(locationLinks)
+	if len(tmp) > 0 {
+		return tmp, true
+	}
+	return tmp, false
 }
 
 func (s *locationServiceImpl) getLocationByUUID(uuid string) (location, bool) {
-	location, found := s.locationsMap[uuid]
-	return location, found
+	val := s.locationsMap.Load()
+	if val != nil {
+		location, found := val.(locationsMap)[uuid]
+		return location, found
+	}
+	return location{}, false
 }
 
 func (s *locationServiceImpl) checkConnectivity() error {
```

Comment:
You can remove the comment. With that you can remove the function, and with that you can simplify the whole healthcheck.